### ISSUE TITLE
Here Be Dragons

### ISSUE THREAD

**Issue by sh0inx** [Iridium-Development/IridiumSkyblock#749]
4.0.6
purpur-2062 (1.20.1)

1) the end spawns the dragon portal at (0,0), which is where the first island spawns
2) the dragon spawns (it probably shouldn't)

![image](https://github.com/Iridium-Development/IridiumSkyblock/assets/24255866/df6dd02d-9dfc-4c08-82eb-3d2d92879177)

we'll probably need to look into cancelling it
``EnderDragonChangePhaseEvent`` is null on first spawn, 
``World.getEnderDragonBattle`` grabs the instance of the dragon battle, which is present regardless of dragon's presence (null if not the end)
``DragonBattle.RespawnPhase`` determines if the dragon is being respawned, we should add a listener to cancel this

``EntitySpawnEvent`` may be something we consider canceling as well

basically we'll need this page: [Spigot API](https://hub.spigotmc.org/javadocs/spigot/org/bukkit/entity/EnderDragon.html)
